### ISSUE TITLE
Custom theme sass file for light mode unexpectedly applied to dark mode

### ISSUE THREAD

**Issue by sun123zxy** [quarto-dev/quarto-cli#3705]
### Bug description

I'm running quarto 1.2.269 on Windows, using sass files to customize HTML output. It seems that custom theme files for light mode are unexpectedly applied to dark mode.

For example, consider this qmd file:

```md
---
title: Test
format:
  html:
    theme:
      light: [flatly, "custom.scss"]
      dark: [darkly]
---
```

and in `custom.scss`:

```scss
/*-- scss:rules --*/
body::after{
    content: "foo";
}
```

I expected to get "foo" in light mode only. However, after rendering, "foo" can be seen in both light and dark mode.

I guess this might have something to do with `bootstrap.min.css` and `bootstrap-dark.min.css` included in `<head>`, since both of them are applied in dark mode.

Apologize if I've missed something obivious. Thanks!

### Checklist

- [X] Please include a minimal, fully reproducible example in a single .qmd file? Please provide the whole file rather than the snippet you believe is causing the issue.
- [X] Please [format your issue](https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us) so it is easier for us to read the bug report.
- [ ] Please document the RStudio IDE version you're running (if applicable), by providing the value displayed in the "About RStudio" main menu dialog?
- [X] Please document the operating system you're running. If on Linux, please provide the specific distribution.

**Comment by dragonstyle** [comment]
You are correct- when the alternate mode is applied, we retain the light stylesheet as well, applying the dark stylesheet as well. This means that situations like this issue are true.  We would ultimately like to take a different approach with dark / light toggling which also does a better job minimizing any flash as users navigate between pages. I'm not 100% sure when we'll get to that, so leaving this open for the time being.

**Comment by kv9898** [comment]
I notice the discussion on flickering for the dark mode (#5900) is relocated here. I noted that this problem has not been addressed for a year. Currently I'm using the following as a fix for my own website. If needed, I can transform this into a pull request, but I think maybe some discussion is good since I'm not entire sure the consequences of my fix. Currently I don't see any visible flickers in my website: https://rubuky.com/



Basically, the gist is to "hide" the content until the dark mode is fully loaded. To do this, we need to adjust `_quarto.yml` for the project to include:

```yaml
format: 
  html:
    include-before-body: 
      - text: |
          <style>
            body {
              visibility: hidden; /* initially hidden until DOM ready to prevent flicker */
            }
          </style>
```

The change above "hides" the page content. 

We then need to change 

https://github.com/quarto-dev/quarto-cli/blob/252bc958e17384c5c473eca50090ee653995d361/src/resources/formats/html/templates/quarto-html.ejs#L190-L194

to

```js
  // Switch to dark mode if need be
  function runToggleColorMode() {
    return new Promise((resolve) => {
      if (hasAlternateSentinel()) {
        toggleColorMode(true);
      } else {
        toggleColorMode(false);
      }
      resolve();
    });
  }

  runToggleColorMode().then(() => {
    setTimeout(() => {
      document.body.style.visibility = "visible";
    }, 50); // Adjust the delay to allow for the dark mode to be loaded
  });
```

This allows the content to be displayed only after the "invisible flicker" has occured, so that we "bypass" the flicker.

**Comment by gordonwoodhull** [comment]
The original issue described here is not a bug; it is by design.

Dark mode for both themes and brands is implemented as light + dark: there is no light-only sass (this goes for themes and brands as well).

We compile to two CSS stylesheets. The light stylesheet is always enabled, and we enable the dark stylesheet via JS when in dark mode.

This allows more concision for customizations common to both modes, and we aren't going to change functionality that users rely on.

In the future we may converge on Bootstrap's color modes #6741, which are more general and don't require enabling/disabling stylesheets dynamically. But we think that's likely to be a breaking change, at the very least requiring a new API with conversions from the current one.

As for the flicker on loading pages in dark mode: although mentioned here, the proper issue is #1325 and was fixed in 1.7.
